src/EditorFeatures/Test/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs

```diff
@@ -55,11 +57,26 @@ internal override IEnumerable<Diagnostic> GetDiagnostics(TestWorkspace workspace
             using (var testDriver = new TestDiagnosticAnalyzerDriver(document.Project, provider))
             {
                 var diagnostics = testDriver.GetAllDiagnostics(provider, document, span);
+                AssertNoAnalyzerExceptionDiagnostics(diagnostics);
+
                 var fixer = providerAndFixer.Item2;
                 var ids = new HashSet<string>(fixer.FixableDiagnosticIds);
                 var dxs = diagnostics.Where(d => ids.Contains(d.Id)).ToList();
                 return GetDiagnosticAndFixes(dxs, provider, fixer, testDriver, document, span, annotation, fixAllActionId);
             }
         }
+
+        /// <summary>
+        /// The internal method <see cref="AnalyzerExecutor.IsAnalyzerExceptionDiagnostic(Diagnostic)"/> does
+        /// essentially this, but due to linked files between projects, this project cannot have internals visible
+        /// access to the Microsoft.CodeAnalysis project without the cascading effect of many extern aliases, so it
+        /// is re-implemented here in a way that is potentially overly agressive with the knowledge that if this method
+        /// starts failing on non-analyzer exception diagnostics, it can be appropriately tuned or re-evaluated.
+        /// </summary>
+        private void AssertNoAnalyzerExceptionDiagnostics(IEnumerable<Diagnostic> diagnostics)
+        {
+            var exceptionDiagnostics = diagnostics.Where(diag => diag.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.AnalyzerException)).ToList();
```

Comment:
I considered that, but since this will potentially be a good spot to attach a debugger on a test that's unexpectedly failing, I thought it would be useful to have the diagnostics stored in a local for easy examination, but I'm not strongly opposed to inlining this.